aggregator/handler/router/with_ack.go

```diff
@@ -60,10 +61,27 @@ func (d message) Bytes() []byte {
 	return d.buffer.Buffer().Bytes()
 }
 
-func (d message) Size() uint32 {
-	return uint32(len(d.Bytes()))
+func (d message) Size() int {
+	// Use the cap of the underlying byte slice in the buffer instead of
+	// the length of the byte encoded.
```

Comment:
Can you also explain a bit more on why the cap instead of the length is the right value to use?